### ISSUE TITLE
Replace `github.action_path` with `env.GITHUB_ACTION_PATH`

### ISSUE THREAD

**Issue by maximilianbraun** [check-spelling/check-spelling#54]
Due to the bug of GitHub Action Runner (https://github.com/actions/runner/issues/716), github.action_path context value is not evaluated into valid path in containers on self-hosted runners. The corresponding environment variable GITHUB_ACTION_PATH always contains valid path even actions running on self-hosted runners, so a PR replaces github.action_path context into GITHUB_ACTION_PATH environment variable should help to fix that issues

**Comment by jsoref** [comment]
Sigh. I guess I'll look through what I have and try to make a release with this fix this week, possibly just this fix and possibly today.

**Comment by maximilianbraun** [comment]
Thanks mate :)

**Comment by jsoref** [comment]
This code area is so fragile, I have a hack for a nektos/act bug (that I fixed in https://github.com/nektos/act/releases/tag/v0.2.22) still hanging around.
And on top of that, I have another hack that was fixed for a second bug I tripped on (that was fixed in
https://github.com/nektos/act/releases/tag/v0.2.35) also hanging around.

I think what I'll do is see if I can remove both of these hacks. (the current version of `act` is [v0.2.46](https://github.com/nektos/act/releases/tag/v0.2.46))

**Comment by jsoref** [comment]
So, 411e63674fa67014213f40d79a70cda292702fe0 is my tentative release thing... I'm not quite ready to release it today (I need to manually review each of the commits and think them over...). One of my tests that I ran today flagged a bug (which I fixed), but that definitely argues against me just shipping w/o more testing.

I've cleaned up a number of commits and am starting to get happy with them.

The fix for this specific bug is included in 5a98d952fcf810f2f1862c9b0d53b33ae6b5f953, although there are somewhat related changes in the following commit (7697e8509beb6596583c08b449bfb4c19fb6b9ba).

@maximilianbraun is it possible for you to test any of these?

It's as simple as using `uses: check-spelling/check-spelling@411e63674fa67014213f40d79a70cda292702fe0` or similar (e.g. `uses: check-spelling/check-spelling@7697e8509beb6596583c08b449bfb4c19fb6b9ba`)

I'm going to let this set of changes run for the next day or so to at least get some feedback from our set of repositories. (We use self-hosted runners, but we don't use them for check-spelling...)
